### PR TITLE
youtube-cleanup: hide comments based on words

### PR DESCRIPTION
**Pull request by BPower0036**
I don't know if the default keyword is a good one.
It only show for which purpose you can use this rule

### CODE REVIEW

File: data/filters/youtube-cleanup.yaml [hunk 1]
```diff
@@ -34,6 +41,11 @@ template: |
   {{#if remove-comment-section}}
   www.youtube.com###comments #contents:remove()
   {{/if}}
+  {{#each keywords as |keyword id|}}
+  {{#if ../remove-specific-comments}}
+  www.youtube.com###comment:has-text({{keyword}}):upward(1)
```
Comment (xvello) [inline data/filters/youtube-cleanup.yaml:46]: I'm not sure `:has-text` is the best operator here. If I understand the doc correctly, it seems like `car` would match `precarious`, but not match `Car` (with an uppercase C).

We ideally would like a case-insensitive match on full words, like the `[prop~="word"]` operator does ([see cheatsheet](https://github.com/DandelionSprout/adfilt/blob/master/Wiki/SyntaxMeaningsThatAreActuallyHumanReadable.md#element-removal-aka-cosmetic-rules-aka-hiding-rules-aka--rules)). Unfortunately, I don't think this operator cannot be applied to an element's text, but we can probably emulate it with a regexp. I'll look into it.

**Review by xvello — COMMENTED** [review]
Thanks for your contribution @BPower0036. This one is less straightforward and needs some more investigation, see below.

File: data/filters/youtube-cleanup.yaml (hunk 1 above)
Comment (BPower0036) [reply data/filters/youtube-cleanup.yaml:43]: You are correct that an operator cannot be applied to an element's text.
So I propose the following rule then:
`###content-text:has-text(/\b(word)\b/i):upward(ytd-comment-thread-renderer)`

Comment (xvello) [reply data/filters/youtube-cleanup.yaml:43]: BTW, you can use the `each` shortform without naming the loop variables:

```
{{#each remove-comments-with-words}}
www.youtube.com###comment:has-text(/\b{{this}}\b/i):upward(1)
{{/each}}
```

Comment (BPower0036) [reply data/filters/youtube-cleanup.yaml:43]: Don't know if the former suggested rule is better or not.

Comment (xvello) [reply data/filters/youtube-cleanup.yaml:43]: The regexp works great, even to match a given phrase (like `great video`) :+1: You don't need the parentheses through.
About the rest of the rule, `#content-text` seems too generic to be future proof. What about prefixing the rule with `ytd-comments` to limit the search to that part of the page?

{{#each remove-comments-with-words}}
www.youtube.com##ytd-comments #content-text:has-text(/\b{{this}}\b/i):upward(ytd-comment-thread-renderer)
{{/each}}

Comment (BPower0036) [reply data/filters/youtube-cleanup.yaml]: The CI keeps failing. I have tried different things, but it keeps failing.
You know how to solve?

File: data/filters/youtube-cleanup.yaml [hunk 2]
```diff
@@ -56,6 +63,10 @@ tests:
       remove-comment-section: true
     output: |
       www.youtube.com###comments #contents:remove()
+  - params:
+      remove-comments-with-words: true
+    output: |
+      www.youtube.com##ytd-comments #content-text:has-text(/\b\b/i):upward(ytd-comment-thread-renderer)
```
Comment (xvello) [inline data/filters/youtube-cleanup.yaml:69]: Your template looks good, but you need to test with an actual word list, not `true`. This should work:
```suggestion
  - params:
      remove-comments-with-words:
        - word1
        - word pair2
    output: |
      www.youtube.com##ytd-comments #content-text:has-text(/\bword1\b/i):upward(ytd-comment-thread-renderer)
      www.youtube.com##ytd-comments #content-text:has-text(/\bword pair2\b/i):upward(ytd-comment-thread-renderer)
```

File: data/filters/youtube-cleanup.yaml (hunk 1 above)
Comment (xvello) [reply data/filters/youtube-cleanup.yaml]: Sorry for the bumpy road @BPower0036, is there improvements to the documentation and tooling I could work on to make the contributor experience easier?

Comment (BPower0036) [reply data/filters/youtube-cleanup.yaml]: @xvello Not that I know, the problem is that I am not good at coding at all. What I can I learned myself

**Review by xvello — APPROVED** [review]
Thanks a lot! Releasing now :hourglass:

**Pull request merged**